integrations/gradle-bloop/src/test/scala/bloop/integrations/gradle/ConfigGenerationSuite.scala

```diff
@@ -46,6 +46,210 @@ abstract class ConfigGenerationSuite {
+    assert(!hasPathEntryName("src/main/resources", resultConfig.project.sources))
+    assert(resultConfig.project.sources.size == 1)
+    assert(resultConfig.project.resources.isDefined)
+    assert(hasPathEntryName("src/main/scala", resultConfig.project.resources.get))
+    assert(resultConfig.project.resources.get.size == 1)
+  }
+
+  @Test def worksWithJavaCompilerAnnotationProcessor(): Unit = {
+    val buildFile = testProjectDir.newFile("build.gradle")
+    testProjectDir.newFolder("src", "main", "scala")
+    writeBuildScript(
+      buildFile,
+      """
+        |plugins {
+        |  id 'bloop'
+        |}
+        |
+        |apply plugin: 'scala'
+        |apply plugin: 'bloop'
+        |
+        |repositories {
+        |  mavenCentral()
+        |}
+        |
+        |dependencies {
+        |  compile group: 'org.scala-lang', name: 'scala-library', version: '2.12.8'
+        |  annotationProcessor "org.immutables:value:2.8.2"
+        |}
+        |
+        """.stripMargin
+    )
+
+    val annotatedSource =
+      """
+        |import java.util.List;
+        |import java.util.Set;
+        |import org.immutables.value.Value;
+
+        |@Value.Immutable
+        |public abstract class FoobarValue {
+        |  public abstract int foo();
+        |  public abstract String bar();
+        |  public abstract List<Integer> buz();
+        |  public abstract Set<Long> crux();
+        |}
+      """
+    val annotatedSourceUsage =
+      """
+        |import java.util.List;
+
+        |public class FoobarValueMain {
+        |  public static void main(String... args) {
+        |    FoobarValue value = ImmutableFoobarValue.builder()
+        |        .foo(2)
+        |        .bar("Bar")
+        |        .addBuz(1, 3, 4)
+        |        .build(); // FoobarValue{foo=2, bar=Bar, buz=[1, 3, 4], crux={}}
+
+        |    int foo = value.foo(); // 2
+
+        |    List<Integer> buz = value.buz(); // ImmutableList.of(1, 3, 4)
+        |  }
+        |}
+      """
+
+    createSource(testProjectDir.getRoot, annotatedSource, "main", "java")
+    createSource(testProjectDir.getRoot, annotatedSourceUsage, "main", "java")
+
+    GradleRunner
+      .create()
+      .withGradleVersion(gradleVersion)
+      .withProjectDir(testProjectDir.getRoot)
+      .withPluginClasspath(getClasspath)
+      .withArguments("bloopInstall", "-Si")
+      .build()
+
+    val projectName = testProjectDir.getRoot.getName
+    val bloopFile = new File(new File(testProjectDir.getRoot, ".bloop"), projectName + ".json")
+
+    val resultConfig = readValidBloopConfig(bloopFile)
+
+    assert(resultConfig.project.resolution.nonEmpty)
+    assert(!hasCompileClasspathEntryName(resultConfig, "org.immutables"))
+    assert(!hasRuntimeClasspathEntryName(resultConfig, "org.immutables"))
+    assert(resultConfig.project.java.isDefined)
+    val processorPath =
+      resultConfig.project.java.get.options.dropWhile(_ != "-processorpath").drop(1)
```

Comment:
@Arthurm1 Thank you, this is the right way of adding support for annotation processors ❤️